staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go

```diff
@@ -25,6 +25,9 @@ var (
 	// Only top-level commands/options setup and the k8s.io/component-base/featuregate/testing package should make use of this.
 	// Tests that need to modify feature gates for the duration of their test should use:
 	//   defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.<FeatureName>, <value>)()
+	//
+	// When used in tests, it must be used for feature gates that are enabled at test start time.
+	// It will not work if the feature gate is being enabled during the test.
```

Comment:
"Will not work" is a pretty strong statement. I suppose you ran into a situation where some test startup code was checking a feature gate, did something based on it, and later the test changed the gate.

But there might also be tests where it's okay because the code that is depending on the feature runs later or simply always checks the feature without any lasting effect.

I think I prefer to not add this comment.